lib/model.js

```diff
@@ -3937,18 +3937,24 @@ class Model {
     // Clone values so it doesn't get modified for caller scope and ignore undefined values
     values = _.omitBy(values, value => value === undefined);
 
-    const changedBefore = this.changed() || [];
-
     options = options || {};
     if (Array.isArray(options)) options = {fields: options};
 
     options = Utils.cloneDeep(options);
     const setOptions = Utils.cloneDeep(options);
     setOptions.attributes = options.fields;
+
+    // We want to keep track of the fields changed via side effects
+    const pushed_changed = this._changed;
```

Comment:
1. No camel case names

2. Why `_changed` is used here, use `changed()`?

Reply:
I think you're right: I could use this.changed()